segmentation violation if config.yml file exists in current directory

# Recreation

I happened to have an existing `config.yml` file in my home directory totally unrelated to bob. Putting

```bash
source <(bob completion)
```

in my `.bashrc` resulted in a segfault.  However, it is more easily recreated by:

```bash
mkdir test-dir
cd test-dir
touch config.yml
bob
```

Expected: usage is printed
Actual:

```
❯ bob_0.4.0_linux_amd64
config file invalid: %!w(<nil>)
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x115e5f1]

goroutine 1 [running]:
github.com/benchkram/bob/cli.glob..func20(0x1fad820?, {0x14660db?, 0x0?, 0x0?})
        /home/runner/work/bob/bob/cli/cmd_root.go:88 +0x1f1
github.com/spf13/cobra.(*Command).execute(0x1fad820, {0xc00003a220, 0x0, 0x0})
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:840 +0x563
github.com/spf13/cobra.(*Command).ExecuteC(0x1fad820)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:974 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:902
github.com/benchkram/bob/cli.Execute()
        /home/runner/work/bob/bob/cli/cli.go:27 +0x4c
main.main()
        /home/runner/work/bob/bob/main.go:14 +0x4c

```

# Analysis

There are two aspects to this issue.

* First, an error reading a config file fails to set the GlobalConfig causing the segmentation fault
* Second, a `nil` error is treated as an error, so if a `config.yml` file exists at all in the current direcotry it will result in the segmentation fault

--- a/cli/config.go
+++ b/cli/config.go
@@ -66,20 +66,8 @@ func readConfig(defaults map[string]interface{}) (*config, error) {
 	for key, value := range defaults {
 		viper.SetDefault(key, value)
 	}
-
-	//Read config from file
-	viper.SetConfigName("config")
-	viper.AddConfigPath(".")
-	err := viper.ReadInConfig()
-	switch err.(type) {
-	case viper.ConfigFileNotFoundError:
-		// fmt.Printf("%s\n", aurora.Yellow("Could not find a config file"))
-	default:
-		return nil, fmt.Errorf("config file invalid: %w", err)
-	}
-
 	c := &config{}
-	err = viper.Unmarshal(c)
+	err := viper.Unmarshal(c)
 	if err != nil {
 		return nil, err
 	}
@@ -90,7 +78,6 @@ func readGlobalConfig() {
 	// Priority of configuration options
 	// 1: CLI Parameters
 	// 2: environment
-	// 2: config.yaml
 	// 3: defaults
 	config, err := readConfig(defaultConfig.AsMap())
 	if err != nil {

